app/helpers/renderers_helper.rb

```diff
@@ -0,0 +1,104 @@
+module RenderersHelper
+  def render_template(template:, subjects: {}, params: {}, snippet_variables: {})
+    template_scope = template_scope_class.new(template: template,
+                                              subjects: subjects,
+                                              params: params)
+    renderer_class.new(template_scope).render(snippet_variables: snippet_variables)
+  end
+
+  def template_scope_class
+    Foreman::Renderers::DatabaseTemplateScope
+  end
+
+  def renderer_class
+    Setting[:safemode_render] ? Foreman::Renderers::SafeModeRenderer : Foreman::Renderers::UnsafeModeRenderer
+  end
+
+  # the code below must be refactored to use SafeModeRenderer and UnsafeModeRenderer
+
+  def methods_scope
+    @methods_scope = template_scope_class.methods_class.new(@host)
+  end
+
+  def method_missing(method, *args, &block)
+    return methods_scope.send(method, *args, &block) if methods_scope.respond_to?(method)
+    super
+  end
+
+  def respond_to_missing?(method, include_private = false)
+    methods_scope.methods.include?(method) || super
+  end
+
```

Comment:
For compatibility reasons, the code below has been copied from the old `Foreman::Renderer` module. We need to refactor it to use the new rendering engine.